pkg/lib/indexer/indexer.go

```diff
@@ -513,38 +523,72 @@ func (i ImageIndexer) ExportFromIndex(request ExportFromIndexRequest) error {
 	}
 
 	var errs []error
+	var g errgroup.Group
+	var mu = &sync.Mutex{}
+
+	concurrentGoroutines := make(chan struct{}, concurrencyLimitForExport)
+
 	for _, bundleImage := range bundles {
-		// try to name the folder
-		folderName, err := dbQuerier.GetBundleVersion(context.TODO(), bundleImage)
-		if err != nil {
+		bundleImage := bundleImage // to ensure every go routine gets a it's own copy of closure variable
+
+		g.Go(func() error {
+			concurrentGoroutines <- struct{}{}
+			defer func() { <-concurrentGoroutines }()
+
+			// get the bundle version to be used in naming the folder
+			bundleVersion, err := dbQuerier.GetBundleVersion(context.TODO(), bundleImage)
+			if err != nil {
+				i.Logger.Errorf("error getting bundle version from image: %s", err)
+				return err
+			}
+			if bundleVersion == "" {
+				// operator-registry does not care about the folder name
```

Comment:
This comment feels odd, why not just describe what we're doing rather than why it is happening? It feels like a comment that would quickly go out of date if another module changed behavior. It also probably means that `bundleVersion` is the wrong variable to set here (either `bundleVersion` should actually be `dirPrefix` or something like that, or we should set the value of the variable further down)

Reply:
I don't think this check is needed in the newly modified code (maybe I am wrong?)

Reply:
ah, you are mostly right. the problem is the case where bundleVersion is not set -- technically version is an optional field on the CSV. in that case, you still need to generate something for the value of the folder.

Reply:
Added the check to generate a random name when empty